build(travis): interrupt the build at the first error message

cipher1024 opened the pull request.

jcommelin reviewed (approved):
Looks good to me

rwbarton commented on scripts/detect_errors.py:
```diff
@@ -0,0 +1,15 @@
+try:
+    for line in sys.stdin:
+        sys.stdout.write(line)
+        if 'error' in line: raise sys.exit()
```
This `raise sys.exit()` is confusing. `sys.exit()` already works by raising an exception, so it's like writing `raise (raise SystemExit)` (if that was valid syntax, which it's not).

I think you actually don't need any `try`/`except` to do this, just delete all those lines and the `raise sys.exit()` and fix the block structure.

rwbarton reviewed (commented):
This is a little more complicated than necessary, but more importantly (and not too surprisingly) it seems to be suffering from the same problem (whatever it is) as the earlier `awk` version: the build was cut off after 10 minutes due to no output.

rwbarton committed:
build(travis): interrupt the build at the first error message

Also make travis_long.sh print its progress messages to stderr.
This sidesteps a mysterious issue where piping the output of
travis_long.sh to another program caused that output to be lost
(probably buffered somewhere?) so Travis would kill the build
after 10 minutes.

cipher1024 committed:
Merge branch 'master' into travis-early-exit
